runtime/sema/type.go

```diff
@@ -5521,6 +5766,12 @@ func checkSubTypeWithoutEquality(subType Type, superType Type) bool {
 			return typedSubType.ExplicitInterfaceConformanceSet().
 				Contains(typedSuperType)
 
+		// An interface type is a supertype of a restricted type if the restricted set contains
+		// that explicit interface type. Once interfaces can conform to interfaces, this should instead
+		// check that at least one value in the restriction set is a subtype of the interface supertype
+		case *RestrictedType:
+			return typedSubType.RestrictionSet().Contains(typedSuperType)
```

Comment:
This is necessary in order to check cases where an attachment is declared `attachment A for I {}` for some interface `I`, and then a value `v` of type `{ T1, T2, ... }` appears in the attach expression `attach A() to v`. `v` conforms to `I` as long as at least one of restrictions in its set is `I`. However, this will need to be updated when interfaces can conform to other interfaces to be a subtype check: i.e. `T1 <: I || T2 >: I || ..` cc @SupunS

Reply:
Maybe include this paragraph as a comment in the code? Or make it into an issue and include the URL to that issue in the code comment?